### ISSUE TITLE
Patching leads to a crash in dl_main

### ISSUE THREAD

**Issue by mehQQ** [lunixbochs/patchkit#16]
Hi, I am trying to patch a program static linked with a library(874KB). However, the program crashed in dl_main even if I only inserted the hello64 sample. I think this may be caused by some segment inserting error?
here is the test program
https://drive.google.com/file/d/0B8WhjUrsG-pDQzM5elBaTWdlVFk/view?usp=sharing
Thanks!

**Comment by lunixbochs** [comment]
Execution trace around the problem (generated with `usercorn -trace ./issue11` and `pr -w250 -m -t issue11.trace issue11.patched.trace`):

[tracediff.txt](https://github.com/lunixbochs/patchkit/files/947981/tracediff.txt)

Notice `rax` is `0x69fe4` on the crashing version, and `0x400040` on the working version. 

Relevant program headers for unpatched:

```
  Type           Offset             VirtAddr           PhysAddr
                 FileSiz            MemSiz              Flags  Align
  PHDR           0x0000000000000040 0x0000000000400040 0x0000000000400040
                 0x00000000000001f8 0x00000000000001f8  R E    8
  LOAD           0x0000000000000000 0x0000000000400000 0x0000000000400000
                 0x0000000000069fe4 0x0000000000069fe4  R E    200000
```

Relevant program headers for patched:

```
  Type           Offset             VirtAddr           PhysAddr
                 FileSiz            MemSiz              Flags  Align
  PHDR           0x0000000000069fe4 0x0000000000469fe4 0x0000000000400040
                 0x0000000000000230 0x00000000000001f8  R E    8
  LOAD           0x0000000000000000 0x0000000000400000 0x0000000000400000
                 0x0000000000069fe4 0x0000000000069fe4  R E    2000
```

Looks like a bug in my PHDR relocation (I move the program header table to the end of the TEXT segment so I can more easily add entries).

**Comment by lunixbochs** [comment]
Should be fixed by 8041540

**Comment by mehQQ** [comment]
I am just trying to patch some vulnerable programs, so I use this tool to insert the patch code. It seems working greatly right now, so many thanks. I will report if I find other bugs. (hope not :P)
